pilot/pkg/networking/core/listener.go

```diff
@@ -1111,18 +1114,20 @@ func buildGatewayListener(opts gatewayListenerOpts, transport istionetworking.Tr
 		filterChains = append(filterChains, &listener.FilterChain{
 			FilterChainMatch: match,
 			TransportSocket:  transportSocket,
+			// Setting this timeout enables the gateway to enhance its resistance against memory exhaustion attacks,
+			// such as slow TLS Handshake attacks.
+			TransportSocketConnectTimeout: durationpb.New(defaultGatewayTransportSocketConnectTimeout),
```

Comment:
Looks this is also useful for sidecars @howardjohn

Reply:
OK, I will add them uniformly